pkg/operator/ceph/object/spec.go

```diff
@@ -138,15 +138,64 @@ func (c *clusterConfig) makeRGWPodSpec() v1.PodTemplateSpec {
 	if c.store.Spec.Gateway.SSLCertificateRef != "" {
 		// Keep the SSL secret as secure as possible in the container. Give only user read perms.
 		userReadOnly := int32(0400)
-		certVol := v1.Volume{
-			Name: certVolumeName,
-			VolumeSource: v1.VolumeSource{
-				Secret: &v1.SecretVolumeSource{
-					SecretName: c.store.Spec.Gateway.SSLCertificateRef,
-					Items: []v1.KeyToPath{
-						{Key: certKeyName, Path: certFilename, Mode: &userReadOnly},
-					}}}}
-		podSpec.Volumes = append(podSpec.Volumes, certVol)
+		secretCertVols := []v1.Volume{
+			{
+				Name: certSecretVolumeName,
+				VolumeSource: v1.VolumeSource{
+					Secret: &v1.SecretVolumeSource{
+						SecretName:  c.store.Spec.Gateway.SSLCertificateRef,
+						DefaultMode: &userReadOnly,
+					},
+				},
+			},
+			{
+				Name: certVolumeName,
+				VolumeSource: v1.VolumeSource{
+					EmptyDir: &v1.EmptyDirVolumeSource{
+						// We don't want a cert on disk, only in memory
+						Medium: v1.StorageMediumMemory,
+					},
+				},
+			},
+		}
+		podSpec.Volumes = append(podSpec.Volumes, secretCertVols...)
+
+		tlsCertPath := path.Join(certSecretDir, "tls.crt")
+		tlsKeyPath := path.Join(certSecretDir, "tls.key")
+		tlsCAPath := path.Join(certSecretDir, "ca.crt")
+		tlsOldPemPath := path.Join(certSecretDir, certFilename)
+
+		tlsCertPemPath := path.Join(certDir, certFilename)
+
+		podSpec.InitContainers = []v1.Container{
+			{
+				Name: "rook-ceph-rgw-cert-pemer",
+				Command: []string{
+					"/bin/bash",
+					"-c",
+					`set -e
```

Comment:
It doesn't feel right to use the script, but I'm not thinking of a better way. If we could simplify the script I would be ok with it. For example, if you only create the init container if it's a `kubernetes.io/tls` secret, then we only need lines 178-182.